app/controllers/api/grade_entry_forms_controller.rb

```diff
@@ -31,6 +31,109 @@ def index
       end
     end
 
+    # create a new grade entry form
+    # Requires:
+    #   :short_identifier
+    # Optional:
+    #   :description, :date, :is_hidden
+    #   grade_entry_items:
+    #     :name, :out_of, :bonus
+    def create
+      if has_missing_params?([:short_identifier])
+        # incomplete/invalid HTTP params
+        render 'shared/http_status', locals: { code: '422', message:
+          HttpStatusHelper::ERROR_CODE['message']['422'] }, status: 422
+        return
+      end
+
+      # check if there is an existing assignment
+      form = GradeEntryForm.find_by_short_identifier(params[:short_identifier])
+      unless form.nil?
+        render 'shared/http_status', locals: { code: '409', message:
+          'Grade Entry Form already exists' }, status: 409
+        return
+      end
+
+      ApplicationRecord.transaction do
+        create_params = params.permit(*DEFAULT_FIELDS, :show_total)
+        create_params[:is_hidden] ||= false
+        create_params[:description] ||= ''
+        new_form = GradeEntryForm.new(create_params)
+        unless new_form.save
+          render 'shared/http_status', locals: { code: '500', message:
+            new_form.errors.full_messages.first }, status: 500
+          raise ActiveRecord::Rollback
+        end
+
+        params[:grade_entry_items]&.each&.with_index do |column_params, i|
+          column_params = column_params.permit(:name, :out_of, :bonus).to_h.symbolize_keys
+          grade_item = GradeEntryItem.new(**column_params, grade_entry_form_id: new_form.id, position: i + 1)
```

Comment:
Prefer the association-based method here. Something like `new_form.grade_entry_items.build(...)`

Reply:
(and in `update` below)